harness/typeCoercion.js

```diff
@@ -0,0 +1,254 @@
+---*/
+
+function getValuesCoercibleToIntegerZero() {
+  var result = [];
+
+  var primitiveValues = [
+    // ToNumber
+    null,
+    false,
+    0,
+    "0",
+
+    // ToInteger: NaN -> +0
+    undefined,
+    NaN,
+    "",
+    "foo",
+    "true",
+
+    // ToInteger: floor(abs(number))
+    0.9,
+    -0,
+    -0.9,
+    "0.9",
+    "-0",
+    "-0.9",
+  ];
+
+  // ToPrimitive
+  primitiveValues.forEach(function(zero) {
+    result.push(zero);
+    result = result.concat(getPrimitiveWrappers(zero, "number"));
+  });
+
+  // Non-primitive values that coerce to 0:
+  // toString() returns a string that parses to NaN.
+  result = result.concat([
+    {},
+    [],
+  ]);
+
+  return result;
+}
+
+function getValuesCoercibleToIntegerOne() {
+  var result = [];
+
+  var primitiveValues = [
+    // ToNumber
+    true,
+    1,
+    "1",
+
+    // ToInteger: floor(abs(number))
+    1.9,
+    "1.9",
+  ];
+
+  // ToPrimitive
+  primitiveValues.forEach(function(value) {
+    result.push(value);
+    result = result.concat(getPrimitiveWrappers(value, "number"));
+  });
+
+  // Non-primitive values that coerce to 1:
+  // toString() returns a string that parses to 1.
+  result = result.concat([
+    [1],
+    ["1"],
+  ]);
+
+  return result;
+}
+
+function getValuesCoercibleToIntegerFromInteger(nominalInteger) {
+  assert(Number.isInteger(nominalInteger));
+  var result = [];
+
+  var primitiveValues = [ nominalInteger ];
+
+  // ToInteger: floor(abs(number))
+  if (nominalInteger >= 0) {
+    primitiveValues.push(nominalInteger + 0.9);
+  }
+  if (nominalInteger <= 0) {
+    primitiveValues.push(nominalInteger - 0.9);
+  }
+
+  // ToNumber: String -> Number
+  primitiveValues = primitiveValues.concat(primitiveValues.map(function(number) { return number.toString(); }));
+
+  // ToPrimitive
+  primitiveValues.forEach(function(value) {
+    result.push(value);
+    result = result.concat(getPrimitiveWrappers(value, "number"));
+  });
+
+  // Non-primitive values that coerce to the nominal integer:
+  // toString() returns a string that parsers to a primitive value.
+  result = result.concat(primitiveValues.map(function(number) { return [number]; }));
+
+  return result;
+}
+
+function getPrimitiveWrappers(primitiveValue, hint) {
+  assert(hint === "number" || hint === "string");
+  var result = [];
+
+  if (primitiveValue != null) {
+    // null and undefined result in {} rather than a proper wrapper,
+    // so skip this case for those values.
+    result.push(Object(primitiveValue));
+  }
+
+  result = result.concat(getValuesCoercibleToPrimitiveWithMethod(hint, function() {
+    return primitiveValue;
+  }));
+  return result;
+}
+
+function getValuesCoercibleToPrimitiveWithMethod(hint, method) {
+  var methodNames;
+  if (hint === "number") {
+    methodNames = ["valueOf", "toString"];
+  } else {
+    methodNames = ["toString", "valueOf"];
+  }
+  return [
+    // precedence order
+    {
+      [Symbol.toPrimitive]: method,
```

Comment:
So, we have a need - and a plan - to flag files using ES6+ features, but how do we do this using a harness file? Maybe using the features flag here in this file for Symbol.toPrimitive? cc @rwaldron 

We don't have this requirement for harness files, but it would be nice to add an info tag with the ToInteger and ToNumber operations. This can be done in a follow up for sure.

Reply:
Some ideas:

1. Just blaze ahead without flagging the features. This is done in this PR for `Symbol.toPrimitive` and in `harness/testAtomics.js`
2. Try to have the JavaScript behave differently if features are available at runtime. This is what I did with `BigInt` in this PR.
3. Harness files declare feature dependencies, and those feature dependencies are implicitly brought in by any test that includes the harness file.

Reply:
@leobalter I think the feature flags for ES6 features are most useful when adding the feature to the test suite initially. At this point, Symbol.toPrimitive is implemented in most engines, and it's difficult to add ever in the future in transpilers, so I'm not sure it's the highest-priority feature flag to get perfectly right.

Reply:
> Just blaze ahead without flagging the features.

This is problematic for consumers of test262 that want to target their test runs while developing specific features. 

> Harness files declare feature dependencies, and those feature dependencies are implicitly brought in by any test that includes the harness file.

This is problematic for consumers of test262 that expect all test files to have the full meta information available in the test file itself. 


--------------------------


> Try to have the JavaScript behave differently if features are available at runtime. This is what I did with BigInt in this PR.

Can you elaborate on this?